src/mail.sh

```diff
@@ -450,6 +455,96 @@ function check_existent_group()
   return 0
 }
 
+# This function removes a given kw_mail group and
+# all of it's references in the data base. Also
+# removes the contacts without group association
+#
+# @group_name: The name of the removed group
+#
+# Returns:
+# returns 0 if successful, non-zero otherwise
+function remove_kw_mail_group()
+{
+  local group_name="$1"
+  local group_id
+
+  validate_group_name "$group_name" || exit_msg 'Error while removing new group'
+
+  check_existent_group "$group_name" && exit_msg $'This group doesnt exist \nError while removing new group'
+
+  group_id="$?"
+
+  remove_group_contacts_association "$group_id" || exit_msg 'Error while removing contacts associations'
+
+  remove_contacts_without_group "$group_id" || exit_msg 'Error while removing contacts without group'
+
+  remove_group "$group_name" || exit_msg 'Error while removing group'
```

Comment:
Don't we have a cascade option that handles this type of situation? For example, remove from the group tables removes all the elements from the associated tables.

@davidbtadokoro